### PR TITLE
Refactor `<sl-select>`

### PR DESCRIPTION
**Pull request by jpzwarte**

### CODE REVIEW

File: packages/components/select/package.json [hunk 1]
```diff
@@ -38,8 +38,7 @@
     "test": "echo \"Error: run tests from monorepo root.\" && exit 1"
   },
   "dependencies": {
-    "@floating-ui/dom": "^1.5.3",
-    "@sl-design-system/popover": "0.1.0",
+    "@sl-design-system/form": "0.0.2",
     "@sl-design-system/shared": "0.2.0",
```
Comment (anna-lach) [inline packages/components/select/package.json:42]: Icon dependency should be added here as well?

File: packages/components/form/src/form-field.stories.ts [hunk 2]
```diff
@@ -81,6 +82,18 @@ export const RadioGroup: Story = {
   }
 };
 
+export const Select: Story = {
```
Comment (anna-lach) [inline packages/components/form/src/form-field.stories.ts:85]: it looks like that `invalid` state needs to look differently comparing to Figma:
image

And currently in the Storybook
image

File: packages/components/select/src/select.stories.ts [hunk 3]
```diff
@@ -78,24 +146,68 @@ export const Basic: StoryObj = {
...
+  }
 };
 
-const options = html`
-  <sl-select-option>🐷 Pig</sl-select-option>
-  <sl-select-option selected>🐨 Koala</sl-select-option>
-  <sl-select-option>🐼 Panda</sl-select-option>
-  <sl-select-option>🦊 Fox</sl-select-option>
-`;
 export const All: StoryObj = {
```
Comment (anna-lach) [inline packages/components/select/src/select.stories.ts:195]: Can you please update `All` story to show `valid` and `invalid` state as well (with the right styling)? :)

image

File: packages/components/select/src/select-option-group.spec.ts [hunk 4]
```diff
@@ -6,25 +6,28 @@ import { SelectOptionGroup } from './select-option-group.js';
...
+      </sl-select-option-group>
+    `);
+  });
+
+  it('should render correctly', () => {
+    expect(el).shadowDom.to.equalSnapshot();
+  });
+
+  it('should have a group role', () => {
+    expect(el).to.have.attribute('role');
```
Comment (Diaan) [inline packages/components/select/src/select-option-group.spec.ts:23]: you don't check what the role is

**Review by Diaan — APPROVED** [review]

File: packages/components/form/src/form-field.stories.ts (hunk 2 above)
Comment (jpzwarte) [reply packages/components/form/src/form-field.stories.ts:85]: Fixed!

**Review by anna-lach — APPROVED** [review]

**Pull request merged**